forge/csrc/passes/erase_inverse_ops.cpp

```diff
@@ -306,37 +311,44 @@ void commute_and_bypass(graphlib::Graph *graph, std::vector<graphlib::Node *> co
             // (1024,) Placing an unsqueeze clone will cause the input to be (1, 1024) which isn't wrong but also is not
             // correct. In this case we shall convert the clone to a reshape which contains the correct number of
             // unsqueezes implicitly.
-            if ((first->op_name() == "unsqueeze" or first->op_name() == "squeeze") and
-                dynamic_cast<graphlib::InputNode *>(graph->node_by_id(operand_edge.producer_node_id)))
+            if ((first->op_name() == "unsqueeze" or first->op_name() == "squeeze"))
             {
                 op->change_op_type("reshape");
-                graphlib::Shape op_shape = graphlib::Shape::create(std::vector<uint32_t>(consumer->shape().size(), 1));
-                auto input = dynamic_cast<graphlib::InputNode *>(graph->node_by_id(operand_edge.producer_node_id));
-
-                for (int i = -1; i >= -(int)input->shape().size(); i--)
+                std::vector<graphlib::Edge> clone_data_edges = graph->operand_data_edges(clone);
+                for (const auto &clone_data_edge : clone_data_edges)
```

Comment:
Let's not use auto if we can, instead use the actual type.

Reply:
Updated